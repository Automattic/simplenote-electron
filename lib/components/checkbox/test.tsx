import Checkbox from './';
import CheckmarkIcon from '../../icons/checkmark';
import CircleIcon from '../../icons/circle';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

it("hasn't had its output unintentionally altered", () => {
  const tree = renderer.create(<Checkbox checked={false}></Checkbox>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the circle icon when unchecked', () => {
  const checkbox = shallow(<Checkbox checked={false}></Checkbox>);
  expect(checkbox.find(CircleIcon)).toHaveLength(1);
  expect(checkbox.find(CheckmarkIcon)).toHaveLength(0);
});

it('renders the checkmark icon when checked', () => {
  const checkbox = shallow(<Checkbox checked={true}></Checkbox>);
  expect(checkbox.find(CircleIcon)).toHaveLength(0);
  expect(checkbox.find(CheckmarkIcon)).toHaveLength(1);
});

it('should call onChange prop when span is clicked', () => {
  const noop = jest.fn();
  const output = renderer.create(<Checkbox onChange={noop}></Checkbox>);
  output.root.findByType('span').props.onClick();
  expect(noop).toHaveBeenCalled();
});
