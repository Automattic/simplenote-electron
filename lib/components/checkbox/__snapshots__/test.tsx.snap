// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hasn't had its output unintentionally altered 1`] = `
<span
  aria-checked={false}
  className="checkbox"
  role="checkbox"
  tabIndex="0"
>
  <span
    aria-hidden="true"
    className="checkbox__icon"
  >
    <svg
      className="icon-circle"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        transform="translate(2 2)"
      >
        <path
          d="m10 0c-5.523 0-10 4.477-10 10s4.477 10 10 10 10-4.477 10-10-4.477-10-10-10zm0 18.5c-4.694 0-8.5-3.806-8.5-8.5s3.806-8.5 8.5-8.5 8.5 3.806 8.5 8.5-3.806 8.5-8.5 8.5z"
        />
      </g>
    </svg>
  </span>
</span>
`;
